src/box/sql/func.c

```diff
@@ -522,6 +522,121 @@ func_trim_str(struct sql_context *ctx, int argc, const struct Mem *argv)
 		ctx->is_aborted = true;
 }
 
+static int
+sql_utf8_symbol_count(const char *str, int byte_len)
+{
+	int symbol_count = 0;
+	int str_end = 0;
+	UChar32 c = 0;
+	while (str_end < byte_len) {
+                U8_NEXT((uint8_t *)str, str_end, byte_len, c);
+                symbol_count++;
+        }
+        return symbol_count;
+}
+
+static char*
+sql_kmp(struct sql_context *ctx, const char *str, const int str_size, 
+	                         const char *key, const int key_size,
+				 int *offset_out, int *pos_out)
+{
+	struct coll *coll = ctx->coll;
+	if (coll == NULL) {
+		char *res = memmem(str, str_size, key, key_size);
+		if (res == NULL) {
+			*pos_out = 0; 
+			*offset_out = str_size;
+		} else {
+			*pos_out = res - str + 1;
+			*offset_out = res - str;
+		}
+		return res;
+	}
+	const int key_chars = sql_utf8_symbol_count(key, key_size);
+	int *prefix_table = calloc(key_chars, sizeof(int));
```

Comment:
Please move `prefix function` out of this function. I mean, it should be a new function. Also, inline code from sql_utf8_symbol_count() to this new function.

Reply:
Should I name the new function `prefix_function` or `prefix_function_new` or `kmp_prefix_function_new`?

Reply:
I think `prefix_function` is fine.